pkg/ir/xds.go

```diff
@@ -0,0 +1,74 @@
+package ir
+
+import (
+	route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+)
+
+// Xds holds the intermediate representation of a Gateway and is
+// used by the xDS Translator to convert it into xDS resources.
+type Xds struct {
+	// Name of the Xds IR.
+	Name string
+	// HTTP listeners exposed by the gateway.
+	Http []HttpListener
+}
+
+// HttpListener holds the listener configuration.
+type HttpListener struct {
+	// Address that the listener should listen on.
+	Address string
+
+	// Port on which the service can be expected to be accessed by clients.
+	Port uint32
+
+	// Hostnames (Host/Authority header value) with which the service can be expected to be accessed by clients.
+	Hostnames []string
+
+	// Tls certificate info. If omitted, the gateway will expose a plain text HTTP server.
+	Tls ServerTLSSettings
+
+	// Routes associated with HTTP traffic to the service.
+	Routes []HttpRoute
+}
+
+// HttpRoute holds the route information associated with the HTTP Route
+type HttpRoute struct {
+	// Matchers define the match conditions for this route.
+	Matchers []route.HeaderMatcher
```

Comment:
`matchers` is not proper English and reads wierd, e.g. `matchers[1].headers...`.

Whereas `matches` is proper English and reads well, e.g. `matches[1].headers...`.

Reply:
moved to matches

Reply:
I still see the field as `matchers`, pls update.